src/main/java/jeresources/profiling/ProfilingExecutor.java

```diff
@@ -18,11 +18,11 @@ public ProfilingExecutor(Profiler profiler) {
         this.executor = Executors.newFixedThreadPool(processors * 2);
     }
 
-    public void addChunkProfiler(DummyWorld dummyWorld, List<Chunk> chunks) {
-        final int dimId = dummyWorld.provider.getDimensionType().getId();
+    public void addChunkProfiler(DummyWorld dummyWorld, List<Chunk> chunks, int dimId) {

```

Comment:
If you change it to a parameter at least keep it final. Otherwise it might do weird things during multi threading.